lib/web/forms/registration.go

```diff
@@ -0,0 +1,71 @@
+package webforms
+
+import (
+	"errors"
+
+	"go-google-scraper-challenge/constants"
+	"go-google-scraper-challenge/helpers"
+	"go-google-scraper-challenge/lib/models"
+
+	validation "github.com/go-ozzo/ozzo-validation/v4"
+	"github.com/go-ozzo/ozzo-validation/v4/is"
+)
+
+type RegistrationForm struct {
+	Email                string `form:"email"`
+	Password             string `form:"password"`
+	PasswordConfirmation string `form:"password_confirmation"`
+}
+
+func (f RegistrationForm) Validate() (valid bool, err error) {
+	err = validation.ValidateStruct(&f,
+		validation.Field(&f.Email, validation.Required, is.EmailFormat),
+		validation.Field(&f.Password, validation.Required),
+		validation.Field(&f.PasswordConfirmation, validation.Required.When(f.Password != ""), validation.By(f.validatePasswordConfirmation(f.Password))),
+	)
+
+	if err != nil {
+		return false, err
+	}
+
+	return true, nil
+}
+
+func (f RegistrationForm) validatePasswordConfirmation(password string) validation.RuleFunc {
+	return func(value interface{}) error {
+		if value.(string) != password && len(password) > 0 {
+			return errors.New("does not match the password")
+		}
+
+		return nil
+	}
+}
+
+func (f RegistrationForm) Save() (*models.User, error) {
+	existingUser, err := models.GetUserByEmail(f.Email)
+	if err == nil && existingUser != nil {
+		return nil, errors.New(constants.UserAlreadyExist)
+	}
+
+	hashedPassword, err := helpers.HashPassword(f.Password)
+	if err != nil {
+		return nil, err
+	}
+
+	user := &models.User{
+		Email:          f.Email,
+		HashedPassword: hashedPassword,
+	}
+
+	userID, err := models.CreateUser(user)
+	if err != nil {
+		return nil, err
+	}
+
+	user, err = models.GetUserByID(userID)
+	if err != nil {
+		return nil, err
+	}
```

Comment:
I think we can skip this part because at last, you're using the user's id to set the current user in the controller.
Or you have a plan for using in the other place?